UM/Settings/ContainerQuery.py

```diff
@@ -1,14 +1,17 @@
-# Copyright (c) 2018 Ultimaker B.V.
+# Copyright (c) 2019 Ultimaker B.V.
 # Uranium is released under the terms of the LGPLv3 or higher.
 
+import collections  # To cache queries.
 import re
-from typing import Any, cast, Dict, Iterable, List, Optional, Type, TYPE_CHECKING
-from UM.Logger import Logger
+from typing import Any, cast, Dict, List, Optional, Tuple, Type, TYPE_CHECKING
 import functools
 
 if TYPE_CHECKING:
     from UM.Settings.ContainerRegistry import ContainerRegistry
 
+# The maximum amount of query results we should cache
+MaxQueryCacheSize = 10000
```

Comment:
Will setting it at 10k ever trigger the mem cleanup (if it doesn't, we might as well drop it all together, right?)

Reply:
I've added all of Ultimaker's printers and all of Creality's printers to initiate many, many queries and it didn't hit this limit. I'll remove the limit then.